ambry-api/src/main/java/com.github.ambry/account/AccountBuilder.java

```diff
@@ -0,0 +1,216 @@
+    this.name = origin.getName();
+    this.status = origin.getStatus();
+    for (Container container : origin.getAllContainers()) {
+      idToContainerMetadataMap.put(container.getId(), container);
+    }
+    this.version = ACCOUNT_METADATA_VERSION_1;
+  }
+
+  /**
+   * Constructor. The builder will not include any {@link Container} information.
+   * @param id The id of the {@link Account} to build.
+   * @param name The name of the {@link Account}. Cannot be {@code null}.
+   * @param status The status of the {@link Account}. Cannot be {@code null}.
+   */
+  public AccountBuilder(short id, String name, AccountStatus status) {
+    this.id = id;
+    this.name = name;
+    this.status = status;
+    this.version = ACCOUNT_METADATA_VERSION_1;
+  }
+
+  /**
+   * Sets the id of the {@link Account} to build.
+   * @param id The id to set.
+   * @return This builder.
+   */
+  public AccountBuilder setId(short id) {
+    this.id = id;
+    return this;
+  }
+
+  /**
+   * Sets the name of the {@link Account} to build.
+   * @param name The name to set.
+   * @return This builder.
+   */
+  public AccountBuilder setName(String name) {
+    this.name = name;
+    return this;
+  }
+
+  /**
+   * Sets the status of the {@link Account} to build.
+   * @param status The id to set.
+   * @return This builder.
+   */
+  public AccountBuilder setStatus(AccountStatus status) {
+    this.status = status;
+    return this;
+  }
+
+  /**
+   * Sets a {@link Container} for the {@link Account} to build. If the builder already has a {@link Container} with
+   * the same id as new {@link Container} to set, the new {@link Container} will replace the existing {@link Container}.
+   * @param container The new {@link Container} to set.
+   * @return This builder.
+   */
+  public AccountBuilder setContainer(Container container) {
+    if (container != null) {
+      idToContainerMetadataMap.put(container.getId(), container);
+    }
+    return this;
+  }
+
+  /**
+   * Removes a {@link Container} in this builder, so that an {@link Account} to build will not have this
+   * {@link Container}. It will be a no-op if no {@link Container} with the id exists in this builder.
+   * @param container The {@link Container} to remove.
+   * @return This builder.
+   */
+  public AccountBuilder removeContainer(Container container) {
+    if (container != null) {
+      short id = container.getId();
+      Container containerToRemove = idToContainerMetadataMap.get(id);
+      if (container.equals(containerToRemove)) {
+        idToContainerMetadataMap.remove(id);
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Sets a list of {@link Container}s for the {@link Account} to build. If any {@link Container} in the list already
+   * exists in the builder, it will be replaced by the one to set.
+   * @param containers A list of {@link Container}s to set.
+   * @return This builder.
+   */
+  public AccountBuilder setContainers(List<Container> containers) {
```

Comment:
Is this method required? It seems like it's the same as calling setContainer multiple times. I could see it potentially being useful if it would clear the existing list of containers and replace them with the provided list